### PR TITLE
Polygon event listener

### PR DESCRIPTION
**Pull request by madduccino**
This fixes the polygons not being clickable bug referenced in #469. It uses the isPointInPath() method, which, instead of defining the shape by top, bottom, right, and left bounds, determines whether the mouse position is a point in the shape's path. The translateToCenter function does not work when this method is used so adjustments for this are made in that function definition.

### CODE REVIEW

**Comment by madduccino** [comment]
I discovered a bug after submitting this pull request. I ended up testing this with two polygons instead of one, and things start behaving very strangely. I think this is because the isPointInPath method only works on the last drawn path. [Here's a little on this issue](https://stackoverflow.com/questions/7984420/html5-canvas-method-ispointinpath-determines-only-the-last-object?rq=1. 

Maybe a better strategy would be to abandon the isPointInPath method and instead determine what the bounds would be for a polygon [here](https://github.com/stevekrouse/WoofJS/blob/65734178f5d84efbbbcf614a4e753997d2f96a42/src/woof.es6.js#L827)?

**Comment by stevekrouse** [comment]
It seems like the problem is that the [polygon `collider` only adds 4 sides](https://github.com/stevekrouse/WoofJS/blob/65734178f5d84efbbbcf614a4e753997d2f96a42/src/woof.es6.js#L1138), when it should add `polygonSides` number of sides

**Comment by stevekrouse** [comment]
My last comment may not have been clear. What I was trying to say is the the `collider` is what's responsible for detecting clicks and also detecting when shapes intersect other sprites. If there's a bug in polygon click-ability, why not fix the `collider` method? In this case, I think the problem is that the [polygon `collider` only adds 4 sides](https://github.com/stevekrouse/WoofJS/blob/65734178f5d84efbbbcf614a4e753997d2f96a42/src/woof.es6.js#L1138), when it should add `polygonSides` number of sides. Does that make sense?

**Comment by madduccino** [comment]
I didn't realize the `collider` was responsible for detecting clicks; I thought it only detected when shapes intersect. Why, then, is an oval clickable and not a polygon?

**Comment by stevekrouse** [comment]
Oh wow, you're totally right. That should definitely be changed. The [`overlap()` method](https://github.com/stevekrouse/WoofJS/blob/2c11757784bcba0cb8e27b7582078d091cdec5bb/src/woof.es6.js#L822) should be defined in terms of SAT.js shapes, and then we can define `over()` in terms of SAT.js methods, such as [`pointInPolygon`](https://github.com/jriecken/sat-js#satpointinpolygonp-poly).

To answer your question about why ovals are clickable, but not polygons... currently the `overlap()` method uses the `bounds()` method, which calculates a shape's bounds based on its `width` and `height`, which polygon's currently don't have implemented. I spent the last 30 minutes trying to find or derive a formula to turn a regular polygon's number of sides and length of side into it's width & height, but I wasn't able to come up with something. If you can find this formula relatively easily, it'd be a quick fix for polygon clickability... but eventually we'd want to do the SAT collider solution because it's more general and precise.

**Comment by madduccino** [comment]
This commit makes polygons clickable by implementing SAT.js methods, as discussed in above comments. It also addresses issue #268 by making colliders more precise.

File: src/woof.es6.js [hunk 1]
```diff
@@ -820,20 +842,58 @@ Woof.prototype.Sprite = function({project = undefined, x = 0, y = 0, angle = 0,
...
-    return !(left > r1.right || right < r1.left || top < r1.bottom || bottom > r1.top);
+    
+    var collider = this.collider();
+    
+    // If collider is a circle, make sure it is inside canvas
+    if (collider.constructor.name == "Circle") {
+      return !(left > collider.pos.x + collider.r || right < collider.pos.x - collider.r || top < collider.pos.y - collider.r || bottom > collider.pos.y + collider.r);
+    }
+    
+    // Otherwise, collider is a polygon. Find out if it's inside canvas by finding min and max points
```
[the remark this answers is not in the document]
Comment (madduccino) [reply src/woof.es6.js]: Would it make sense to...

Add this method to the Woof object:

`thisContext.collider = (left, right, top, bottom) => {
    return new SAT.Polygon(new SAT.Vector(), [
    new SAT.Vector(left, top),
    new SAT.Vector(left, bottom),
    new SAT.Vector(right, bottom),
    new SAT.Vector(right, top)
  ]);
}
`
And then change the Sprite overlap method to:

`this.overlap = ({left, right, top, bottom}) => {
 
    var projectCollider = this.project.collider(left, right, top, bottom);
    var objectCollider = this.collider();
    var response = new SAT.Response();
  
    if (objectCollider.constructor.name == "Circle") {
      return SAT.testPolygonCircle(projectCollider, objectCollider, response)
    } else {
      return SAT.testPolygonPolygon(projectCollider, objectCollider, response)
   }
`

File: docs/index.html [hunk 2]
```diff
@@ -706,7 +706,7 @@
                             
                             {
                                 url: "./images/boss-ship.png",
-                                code: 'var bossShip2 = new Image({\n  url: "./docs/boss-ship/explosion.png",\n  width: 100, \n  height: 100,\n})',
```
[the remark this answers is not in the document]
Comment (madduccino) [reply docs/index.html]: I had committed this branch and created a pull request on Github. Changes were subsequently made on another branch and merged with the master. So when I revisited this branch, it was no longer current. Do you have any recommendations on how to handle this in the future?

File: src/woof.es6.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (stevekrouse) [reply src/woof.es6.js]: Yeah, basically. I'm not sure why you have to pass in the `left, right, top, bottom` parameters to the Woof collider. Doesn't it know how big it is via the `width` and `height` variables?

File: docs/index.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (stevekrouse) [reply docs/index.html]: Ah, I see how that could be confusing. Git branches were built to handle exactly this problem. You can work on multiple independent branches at the same time, and then when they're done, you can merge them together and git handles the updating of old branches, etc.

In short, you should've ignored the fact that this branch was no longer current, and everything would've worked out great. Does that makes sense?

[the remark this answers is not in the document]
Comment (madduccino) [reply docs/index.html]: I think so!

**Pull request merged**